app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,72 @@
+require 'rack-flash'
+
+class TweetsController < ApplicationController
+
+  use Rack::Flash
+
+  get '/tweets/new' do
+    redirect to '/login' unless logged_in?
+
+    erb :'tweets/create_tweet'
+  end
+
+  get '/tweets/:id' do
+    redirect to '/login' unless logged_in?
+
+    @tweet = Tweet.find(params[:id])
+    erb :'/tweets/show_tweet'
+  end
+
+  get '/tweets/:id/edit' do
+    redirect to '/login' unless logged_in?
+
+    @tweet = Tweet.find(params[:id])
+
+    if @tweet.user.id != current_user.id
+      flash[:notice] = "You cannot edit another user's tweet!"
+      redirect to "/tweets/#{@tweet.id}"
+    end
+
+    erb :'/tweets/edit_tweet'
+  end
+
+  get '/tweets' do
+    redirect to '/login' unless logged_in?
+
+    @tweets = Tweet.all
+    erb :'/tweets/tweets'
+  end
+
+  post '/tweets' do
+    @tweet = Tweet.new(content: params[:content], user_id: current_user.id)
+
+    unless @tweet.save
+      flash[:notice] = "Invalid tweet. Please try again."
+      redirect to '/tweets/new'
+    end
+
+    redirect to "/tweets/#{@tweet.id}"
+  end
+
+  patch '/tweets/:id' do
+    @tweet = Tweet.find(params[:id])
+
+    # ActiveRecord validation not triggering for some reason so checking manually instead. Pending.
+    if params[:content].empty?
+      flash[:notice] = "Invalid content. Please try again."
+      redirect to "/tweets/#{@tweet.id}/edit"
+    elsif @tweet.update(content: params[:content])
+      redirect to "/tweets/#{@tweet.id}"
+    else
+      redirect to '/tweets'
+    end
+  end
+
+  delete '/tweets/:id/delete' do
```

Comment:
Prefer to find tweets that are already only associated to the `current_user` than to manually check the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
tweet = current_user.tweets.find_by(id: params[:id])
if tweet && tweet.destroy
  redirect "/tweets"
else
  redirect "/tweets/#{tweet.id}"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully, simply returning nil if the user is trying to delete a tweet that they do not have access to.